plugin/trino-kafka-event-listener/src/main/java/io/trino/plugin/eventlistener/kafka/KafkaEventPublisher.java

```diff
@@ -0,0 +1,149 @@
+
+import io.airlift.log.Logger;
+import io.trino.plugin.eventlistener.kafka.metadata.EnvMetadataProvider;
+import io.trino.plugin.eventlistener.kafka.metadata.MetadataProvider;
+import io.trino.plugin.eventlistener.kafka.metadata.NoOpMetadataProvider;
+import io.trino.plugin.eventlistener.kafka.metrics.KafkaEventListenerJmxStats;
+import io.trino.plugin.eventlistener.kafka.producer.KafkaProducerFactory;
+import io.trino.plugin.eventlistener.kafka.producer.SSLKafkaProducerFactory;
+import io.trino.spi.eventlistener.QueryCompletedEvent;
+import io.trino.spi.eventlistener.QueryCreatedEvent;
+import org.apache.avro.generic.GenericRecord;
+import org.apache.kafka.clients.producer.KafkaProducer;
+import org.apache.kafka.clients.producer.ProducerRecord;
+
+import java.util.Map;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.TimeUnit;
+
+public class KafkaEventPublisher
+{
+    private static final Logger LOG = Logger.get(KafkaEventPublisher.class);
+
+    private final KafkaProducer<String, GenericRecord> kafkaProducer;
+    private final KafkaRecordBuilder kafkaRecordBuilder;
+    private final KafkaEventListenerJmxStats stats;
+
+    KafkaEventPublisher(KafkaEventListenerConfig config, KafkaProducerFactory producerFactory, KafkaEventListenerJmxStats stats)
+            throws Exception
+    {
+        this.stats = stats;
+
+        String startedTopic = config.getCreatedTopicName().orElse("");
+        String completedTopic = config.getCompletedTopicName().orElse("");
+
+        Map<String, String> configOverrides = config.getKafkaClientOverrides();
+        LOG.info("Kafka Event Listener: Creating Kafka publisher (SSL=%s) for topics: %s/%s with excluded fields: %s and kafka config overrides: %s",
+                producerFactory instanceof SSLKafkaProducerFactory, startedTopic, completedTopic, config.getExcludedFields(), configOverrides);
+        kafkaProducer = producerFactory.producer(configOverrides);
+        checkConnectivityToBrokers(config.getPublishCreatedEvent() ? startedTopic : completedTopic, config.getRequestTimeout());
+        kafkaRecordBuilder = new KafkaRecordBuilder(startedTopic, completedTopic, config.getExcludedFields(), metadataProvider(config));
+        LOG.info("Kafka Event Listener: Successfully created Kafka publisher.");
+    }
+
+    private void checkConnectivityToBrokers(String topic, int requestTimeout)
+            throws Exception
+    {
+        try {
+            LOG.debug("Kafka Event Listener: checking connectivity to brokers (fetching partitions for topic=%s).", topic);
+            CompletableFuture<Void> future = CompletableFuture.runAsync(() -> kafkaProducer.partitionsFor(topic));
+            future.get(requestTimeout, TimeUnit.MILLISECONDS);
+            LOG.debug("Kafka Event Listener: connectivity check succeeded.");
+        }
+        catch (Exception e) {
+            LOG.error("Kafka Event Listener: connectivity check failed: %s", e.getMessage());
```

Comment:
Extra error message here is not required. It would be double reported (log + exception) which is confusing when browsing logs